Add PDF reporting task

neilvanbeinum opened the pull request:
[Trello Card](https://trello.com/c/JIT8o1Tn/90-pdf-report-for-natalie-baron-com)

This task produces a CSV of counts of published PDFs over time, grouped
by the owning organisation. This is for the Content Operating model team.

Also see (https://github.com/alphagov/manuals-publisher/pull/814) and (https://github.com/alphagov/whitehall/pull/3036).

There was a timeout problem on the integration environment when accessing a certain document category from publishing-api (on which this script relies). This was addressed in PR https://github.com/alphagov/publishing-api/pull/818.

h-lame commented on lib/tasks/report.rake:
```diff
@@ -60,6 +60,117 @@ namespace :report do
+          :content_id,
+          :state_history,
+        ],
+        per_page: 100000,
```
I don't know if it'll help the timeout issues you were seeing, but I reckon we could reuse the Paginator in the `content_operating_model` report below to avoid doing this (and reduce load on publishing-api trying to return all 30,000 DFID documents in one go).

neilvanbeinum replied:
Good shout, thanks! I've extracted it to a class https://github.com/alphagov/specialist-publisher/pull/1007/commits/20df4318d88361e0faf5c92fb2030b5d5d6d7a56

neilvanbeinum committed:
Extract a document pagination class for reporting

This class is also to be used in an upcoming pdf reporting task, so it
suits being put in an external file.

Also:
  - Modified the constructor ,allowing different report tasks to pass in
  custom content item params.

neilvanbeinum committed:
Add a PDF reporting task for the content operating model team

This task produces a CSV file of counts of published PDFs associated with
organisations within certain time periods.
